odoo/fields.py

```diff
@@ -2224,7 +2224,17 @@ def _related_domain(self):
 
     def _description_domain(self, env):
         if self.check_company and not self.domain:
-            return "['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
+            if self.comodel_name == "res.users":
+                """ For res_users, we need to ensure the company is in the
+                allowed companies of the user.
+
+                But we can only do so when the field is required.  Otherwise,
+                the domain would be void and return nothing.
+                """
+                if env[self.model_name]._fields['company_id'].required:
```

Comment:
what about the case it's not required? this needs an `else`, right?

Reply:
then it falls back on the generic case ;).

Reply:
Edit, fixed ;).